ts/components/CalendarsListContainer.tsx

```diff
@@ -174,30 +91,81 @@ class CalendarsListContainer extends React.PureComponent<Props, State> {
           ResourceError
         > = pot.toError(pot.none, fetchError);
-        this.setState(
-          {
-            calendarsByAccount
-          },
-          () => this.props.onCalendarsLoaded()
-        );
+        setCalendarsByAccount(calendarsByAccount);
+        onCalendarsLoaded();
       });
-  };
-}
-
-const mapStateToProps = (state: GlobalState) => ({
-  defaultCalendar: state.persistedPreferences.preferredCalendar
-});
-
-const mapDispatchToProps = (dispatch: Dispatch) => ({
-  preferredCalendarSaveSuccess: (calendar: Calendar) =>
-    dispatch(
-      preferredCalendarSaveSuccess({
-        preferredCalendar: calendar
-      })
-    )
-});
+  }, [onCalendarsLoaded]);
+
+  const mapData = useCallback(
+    (data: ReadonlyArray<Calendar>) =>
+      data.map((item: Calendar) => ({
+        id: item.id,
+        value: item.title,
+        disabled: !item.allowsModifications
+      })),
+    []
+  );
+
+  const onPressRadio = useCallback(
+    (value: string) => {
+      const calendar =
+        pot.isSome(calendarsByAccount) &&
+        calendarsByAccount.value
+          .flatMap(section => section.data)
+          .find(cal => cal.id === value);
+      if (calendar) {
+        const isDefaultCalendar =
+          defaultCalendar && calendar.id === defaultCalendar.id;
+        setSelectedCalendar(calendar);
+        if (isDefaultCalendar && onCalendarRemove) {
+          onCalendarRemove();
+        } else {
+          onCalendarSelected(calendar);
+        }
+      }
+    },
+    [calendarsByAccount, defaultCalendar, onCalendarRemove, onCalendarSelected]
+  );
+
+  useEffect(() => {
+    fetchCalendars();
+  }, [fetchCalendars]);
+
+  return (
+    <ContentWrapper>
+      {pot.isError(calendarsByAccount) && (
+        <>
+          <Body>{mapResourceErrorToMessage(calendarsByAccount.error)}</Body>
+          <ButtonSolid
+            onPress={fetchCalendars}
+            label={I18n.t("global.buttons.retry")}
+            accessibilityLabel={I18n.t("global.buttons.retry")}
+          />
+        </>
+      )}
+
+      {pot.isSome(calendarsByAccount) && (
+        <>
+          {calendarsByAccount.value.map((section, index) => (
+            <React.Fragment key={index}>
+              <ListItemHeader label={section.title} />
+              <RadioGroup<string>
+                type="radioListItem"
+                key={`radio_group_${index}`}
+                items={mapData(section.data)}
+                selectedItem={selectedCalendar?.id}
+                onPress={onPressRadio}
+              />
+              {/* not show the end spacer if the element is the last */}
+              {index < calendarsByAccount.value.length - 1 && (
+                <VSpacer size={24} />
+              )}
+            </React.Fragment>
+          ))}
+        </>
+      )}
+    </ContentWrapper>
+  );
+};
 
-export default connect(
-  mapStateToProps,
-  mapDispatchToProps
-)(CalendarsListContainer);
+export default CalendarsListContainer;
```

Comment:
Since this component is used as a child of a state-changing component, it might make sense to wrap it with a memo so that it only re-renders when its props change.

Reply:
yess! Thanks for the suggestion!